### PR TITLE
Add rclone

### PR DESCRIPTION
**Pull request by NyaMisty**
rclone can be used in the following scenarios:
1. upload files downloaded by others
2. download files from drives
3. upload all sorts of thing including torrents and backups

so it linked to every data volume ;)

As rclone's web gui is session-less, browser stores plain password in localStorage to do auth, which will then conflict with traefik's default auth. So I make the rclone's password to be rclone:rclone fixed (so other components can call rclone without storing real password), and rewrote the authorization header in traefik. User can login rclone with their traefik password for both http basicAuth and rclone webgui's auth.

### CODE REVIEW

File: services/rclone.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    environment:
+      - USER_ID=${PUID}
+      - GROUP_ID=${PGID}
+      - TZ=${TZ}
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.services.rclone-seedbox.loadbalancer.server.port=5572"
+      - "traefik.http.routers.rclone.rule=Host(`rclone.${TRAEFIK_DOMAIN}`)"
+      - "traefik.http.routers.rclone.middlewares=common-auth@file, rclone-auth@docker"
+      - "traefik.http.middlewares.rclone-auth.headers.customrequestheaders.Authorization=Basic cmNsb25lOnJjbG9uZQ=="
```
Comment (jfroment) [inline services/rclone.yaml:25]: Could we avoid hardcoded sensible values?

File: traefik/custom/middlewares.yaml [hunk 2]
```diff
@@ -12,6 +12,6 @@ http:
         referrerPolicy: "same-origin"
         featurePolicy: "vibrate 'self'; geolocation 'self'; midi 'self'; notifications 'self'; push 'self'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; speaker 'none'; vibrate 'self'; fullscreen 'self'"
 
-        frameDeny: true
+        #frameDeny: true
```
Comment (jfroment) [inline traefik/custom/middlewares.yaml:15]: Why?

**Comment by NyaMisty** [comment]
Sorry I pushed my customization accidentally to this branch, I'll rebase it

**Review by jfroment — CHANGES_REQUESTED** [review]
Hi! Thanks for your inputs!
In order to keep this repository "clean" about what has been done, could you please take into consideration the following items:

- Could you rebase on latest master? Since v1.4, services.conf are now per-user. Also, all new services should be declared here, but now it is services.conf.sample.
- Could you detail more the exhaustive content of the changes in your PR? I see there is not only rclone but also QBitTorrent EE, for example.

**Comment by NyaMisty** [comment]
Yep, the pr should only contains "add Rclone"  commit.
other commits are for my own.

File: services/rclone.yaml (hunk 1 above)
Comment (NyaMisty) [reply services/rclone.yaml:25]: The Rclone webui's login feature is actually storing one's username and password, then authenticate by letting JavaScript pass the authorization header, which conflicts with traefik's authentication.
So instead, I hardcoded rclone's password, and user can use traefik's password to login rclone's webui.

Comment (NyaMisty) [reply services/rclone.yaml:25]: The Rclone webui's login feature is actually storing one's username and password, then authenticate by letting JavaScript pass the authorization header, which conflicts with traefik's authentication.
So instead, I hardcoded rclone's password, and user can use traefik's password to login rclone's webui.

**Comment by jfroment** [comment]
Closed due to inactivity and v2 release which breaks how services are handled and provides means to add custom services.

**Pull request closed without merging**